### PR TITLE
migrate influxdb-cpp to conan v2

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
This conan v2 migration PR was generated by automatoc script version 10.
It uses simple find and replace technics. Feel free to extend it.

### CODE REVIEW

**Review by uilianries — CHANGES_REQUESTED** [review]
Please, take into account the tools guidelines for your script

[docs.conan.io/en/latest/reference/conanfile/tools.html](https://docs.conan.io/en/latest/reference/conanfile/tools.html)

**Review by intelligide — DISMISSED** [review]

**Review by danimtb — DISMISSED** [review]

File: recipes/influxdb-cpp/all/conanfile.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import os
 
-from conans import ConanFile, tools
+from conan import ConanFile, tools
```
Comment (danimtb) [inline recipes/influxdb-cpp/all/conanfile.py:3]:
```suggestion
from conan import ConanFile
from conan.tools.files import get
```

File: recipes/influxdb-cpp/all/test_package/conanfile.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import os
 
-from conans import ConanFile, CMake, tools
+from conan import ConanFile, tools
+from conans import CMake
```
Comment (danimtb) [inline recipes/influxdb-cpp/all/test_package/conanfile.py:4]:
```suggestion
from conan import ConanFile
from conan.tools.build import cross_building
from conan.tools.cmake import CMake, CMakeToolchain
```

File: recipes/influxdb-cpp/all/test_package/conanfile.py [hunk 3]
```diff
@@ -13,6 +14,6 @@ def build(self):
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self.settings):
+        if not tools.build.cross_building(self, self.settings):
```
Comment (danimtb) [inline recipes/influxdb-cpp/all/test_package/conanfile.py:17]:
```suggestion
        if not cross_building(self, self.settings):
```

File: recipes/influxdb-cpp/all/test_package/conanfile.py [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import os
```
Comment (uilianries) [inline recipes/influxdb-cpp/all/test_package/conanfile.py:1]: The test package is mixing V1 with V2, also, self.setting should not be passed to cross_building anymore.

**Pull request closed without merging**